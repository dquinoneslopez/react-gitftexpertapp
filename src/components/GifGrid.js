import React from 'react'
import { useFecthGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
    
    const { data:images, loading } = useFecthGifs(category);
       
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>
            
            { loading && <p className="animate__animated animate__flash">Loading</p> }
            
            <div className="card-grid animate__animated animate__fadeIn">
                {
                    images.map(img => <GifGridItem key={ img.id } {...img}/>)
                }
            </div> 
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

