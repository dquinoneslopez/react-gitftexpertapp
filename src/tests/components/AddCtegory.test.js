import React from 'react'
import { shallow } from 'enzyme'
import { AddCategory } from '../../components/AddCategory'
import '@testing-library/jest-dom'

describe('Pruebas en <AddCategory/>', () => {
    
    // const setCategories = () => {};
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });
    
    test('Debe mostrar el componente correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
        
    })
    
    test('Debe cambiar la caja de texto', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', { target: { value } });
        expect(wrapper.find('p').text().trim()).toBe(value);
        
    })
    
    test('No debe postear la informaciÃ³n con submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(setCategories).not.toHaveBeenCalled();
        
    })
    
    test('Debe llamar el setCategories y limpiar la caja de texto', () => {
        
        const value = 'Hola mundo';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        expect(wrapper.find('input').prop('value')).toBe('');
        
    })
    
    
    
})
