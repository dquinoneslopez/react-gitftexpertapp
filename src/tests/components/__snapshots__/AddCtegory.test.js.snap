// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
